app/src/androidTest/java/edu/uwstout/p2pchat/room/RoomTesting.java

```diff
@@ -0,0 +1,240 @@
+
+        dao.deletePeers(peer);
+
+        assertEquals(dao.getPeers().size(), 1);
+        assertEquals(dao.getPeers().get(0).macAddress, "7c:4f:87:2c:56:d5");
+        assertEquals(dao.getPeers().get(0).nickname, "Armin Arlert");
+
+        dao.deletePeers(peer2);
+
+        assertEquals(dao.getPeers().size(), 0);
+    }
+
+    @Test
+    public void createAndDestroyMessage()
+    {
+        DataAccessObject dao = database.dataAccessObject();
+
+        Peer peer = new Peer("7a:a0:83:04:03:60");
+        peer.nickname = "Eren Jeager";
+        database.dataAccessObject().insertPeers(peer);
+
+        Peer peer2 = new Peer("7c:4f:87:2c:56:d5");
+        peer2.nickname = "Armin Arlert";
+        database.dataAccessObject().insertPeers(peer2);
+
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).size(), 0);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).size(), 0);
+
+        Date now = Calendar.getInstance().getTime();
+
+        Message message = new Message(peer.macAddress);
+        message.timestamp = now;
+        message.mimeType = "text/message";
+        message.sent = true;
+        message.content = "Historia is best girl, don't you agree Armin?";
+        dao.insertMessages(message);
+
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).size(), 1);
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).get(0).content, message.content);
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).get(0).timestamp, message.timestamp);
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).get(0).mimeType, message.mimeType);
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).get(0).macAddress,
+                message.macAddress);
+        assertEquals(dao.getMessagesFromPeer(peer.macAddress).get(0).sent, message.sent);
+
+        Message message2 = new Message(peer2.macAddress);
+        message2.timestamp = now;
+        message2.mimeType = "text/message";
+        message2.sent = false;
+        message2.content = "What about Mikasa? She's head over heels for you.";
+        dao.insertMessages(message2);
+
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).size(), 1);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).get(0).content, message2.content);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).get(0).timestamp,
+                message2.timestamp);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).get(0).mimeType, message2.mimeType);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).get(0).macAddress,
+                message2.macAddress);
+        assertEquals(dao.getMessagesFromPeer(peer2.macAddress).get(0).sent, message2.sent);
+
+        Message message3 = new Message(peer.macAddress);
+        message3.timestamp = now;
+        message3.mimeType = "text/message";
+        message3.sent = true;
+        message3.content = "But Historia is a literal Queen! ばか!";
```

Comment:
We support Japanese characters in the database?

Reply:
Yes